### DISCUSSION TITLE
Remove pure black from image and halos

### DISCUSSION THREAD

**Discussion by amenrio** [ImageMagick/ImageMagick#7675] [Help] [answered]
### ImageMagick version

7.1.1-39

### Operating system, version and so on

Rocky Linux 9.3 

### Description

Hello, I'm testing how to clean some .exr, mainly remove pure blacks arround the red thing, this is the source image `example1.exr`

Due to some NDA's, here is a representation of what the output image looks like

![example1](https://github.com/user-attachments/assets/6959c7ba-8c4e-49fd-b838-6c78ec1457c3)

I tried this

```bash
magick example1.exr -transparent black example1_transparent.exr
```
![example1_transparent](https://github.com/user-attachments/assets/e7796c5a-f523-4056-9fcb-a5de7fb6a080)

> checker background put for ease of view, image is and needs to be transparent and with alpha

I've tried some examples but i cant get rid of the black lines (halos?) without tampering with other colors in the image (The rest must be intact)

EDIT:
Using `fuzz` doesn't work because it starts to remove from the center before it get removed from the outside lines

Since it is separated from the black mass of pixels i think there must be a way to delete only those parts. But I'm stuck here.

Thank you!

**Comment by fmw42** [comment]
Try adding -fuzz XX% to your command before -transparent.  Increase the fuzz value from 0% by as much as needed to remove the shadow lines.

**Reply by amenrio** [reply]
Hi! Thank you for your answer, I've updated the issue since the example wasn't the best, there is some black in the part of the image that I don't want to remove, so before `fuzz` can get rid of the halos it starts to remove also from the center of the image.

**Comment by snibgo** [comment, marked as answer]
"-transparent Black" will change the transparency only where the pixel is fully opaque. Pixels that are black but only partly transparent will not change.

Where redBlob.png is my name for your image, the following makes a copy of the image, and removes the transparency from that. Then `-fx` changes the alpha channel to be the minimum of what it was and the alpha of the cloned version.

```
magick redBlob.png ( +clone -alpha off -transparent Black ) -channel A -fx min(u,v) +channel x.png
```
There may be a faster way to get the same result.

**Reply by amenrio** [reply]
Works flawlessly! Thank you so much!
